locations/spiders/orlen_paczka_pl.py

```diff
@@ -0,0 +1,34 @@
+from scrapy import Request, Spider
+
+from locations.hours import DAYS_PL, OpeningHours
+from locations.items import Feature
+
+
+class OrlenPaczkaPLSpider(Spider):
+    name = "orlen_paczka_pl"
+    item_attributes = {"brand": "Orlen Paczka", "brand_wikidata": "Q110457879"}
+    start_urls = ["https://ruch-osm.sysadvisors.pl/api.php"]
+    allowed_domains = ["ruch-osm.sysadvisors.pl"]
+
+    def start_requests(self):
+        for url in self.start_urls:
+            yield Request(
```

Comment:
I'm on my phone, so can't confirm. But FormRequest should probably be used here. It adds the headers automatically, and converts the data and does post by default.

The following should work:
```suggestion
yield FormRequest(url=url, formdata={"a":"f","s":0})

Reply:
That's what I thought too but it doesn't work because FormRequest doesn't want to convert integer 0. From vague memory, Scrapy doesn't like URL encoded form data?

Reply:
Probably wants a string, "0"

Reply:
> Probably wants a string, "0"

The problem then is that the server ends up receiving a body of:

`{"a":"f","s":"0"}`

Instead of:

`{"a":"f","s":0}`

Treating the 0 as a string causes the server to return no results.

Reply:
Ahh I wasn't paying attention, this isn't actually a Form request, the header is a lie. It's a Json POST request.

`yield JsonRequest(url=url, data={"a": "f", "s": 0})`

The special classes, `JsonRequest` and `FormRequest` do 2 main things.

1. They automatically set the headers, which I think is polite but often not required (or there are times when sites require other or even wrong herders).
2. Automatically populate the body. In the case of Json it's probably just json.dumps, but it's nicer to see and alter a dict than a string containing json.

They also automatically change them to POST requests when there is `data` or `formdata`.